### ISSUE TITLE
[ModernBERT] Training Loss 0.0 / Grad NaN caused by literal special tokens in pre-tokenized input

### ISSUE THREAD

**Issue by lxmliu2002** [ICTMCG/SCM#4]
I encountered a silent failure case where the training loss was constantly 0.0 and grad_norm was NaN immediately from step 1.

{'loss': 0.0, 'grad_norm': nan, 'learning_rate': 2e-05, 'epoch': 0.01}

**Comment by Liesy** [comment]
This is indeed a very strange issue.
Did you use the ModernBERT tokenizer to properly process all your training data?
Literal special tokens usually only appear when the input wasn’t fully tokenized by the official tokenizer or when some texts were pre-tokenized incorrectly.
